vercel/resource_deployment.go

```diff
@@ -303,15 +303,40 @@ func (r resourceDeployment) Read(ctx context.Context, req tfsdk.ReadResourceRequ
 	}
 }
 
-// Update is a noop as it is not possible to update an existing deployment. Instead, all
-// attributes must be set to force recreation.
-// This method has to exist, however, to satisfy the resource interface.
+// Update updates the deployment state.
+// Note that only the `delete_on_destroy` field is updatable, and this does not affect Vercel. So it is just a case
+// of setting terraform state.
 func (r resourceDeployment) Update(ctx context.Context, req tfsdk.UpdateResourceRequest, resp *tfsdk.UpdateResourceResponse) {
-	// Nothing to do here - we can't update deployments
+	var plan Deployment
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		resp.Diagnostics.AddError(
+			"Error getting deployment plan",
+			"Error getting deployment plan",
```

Comment:
nothing else worth adding here?

Reply:
I copy-pasted from the `Create` function. Might have a play and see what the output looks like 🤔. I suspect it might be one of those errors that's really hard to actually see, just because the schema validation kicks in before this point.

Reply:
yeah it's worth figuring out if the errors raised here are actually useful for output, or if they're pretty much just flags to say "stop now, it didn't work"